microshift.sh

```diff
@@ -0,0 +1,141 @@
+fi
+
+run_preflight_checks ${BUNDLE_TYPE}
+rm -fr ${INSTALL_DIR} && mkdir ${INSTALL_DIR}
+
+sudo virsh destroy ${SNC_PRODUCT_NAME} || true
+sudo virsh undefine --nvram ${SNC_PRODUCT_NAME} || true
+sudo rm -fr /var/lib/libvirt/images/${SNC_PRODUCT_NAME}.qcow2
+sudo rm -fr /var/lib/libvirt/images/microshift-installer-*.iso
+
+
+# Generate a new ssh keypair for this cluster
+# Create a 521bit ECDSA Key
+rm id_ecdsa_crc* || true
+ssh-keygen -t ecdsa -b 521 -N "" -f id_ecdsa_crc -C "core"
+
+# This requirement is taken from https://github.com/openshift/microshift/blob/main/scripts/image-builder/configure.sh
+# Also https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/composing_a_customized_rhel_system_image/installing-composer_composing-a-customized-rhel-system-image
+# list out the dependencies and usecase.
+# lorax packages which install mkksiso is required for embedding kickstart file to iso file
+# podman package is required to run the ostree-container to serve the rpm-ostree content
+# createrepo package is required to create localrepo for microshift and it's dependenices
+# yum-utils package is required for reposync utility to synchronize packages of a remote DNF repository to a local directory
+function configure_host {
+    sudo dnf install -y git osbuild-composer composer-cli ostree rpm-ostree \
+      cockpit-composer cockpit-machines bash-completion lorax podman \
+      yum-utils createrepo
+    sudo systemctl start osbuild-composer.socket
+    sudo systemctl start cockpit.socket
+    sudo firewall-cmd --add-service=cockpit
+}
+
+function enable_repos {
+    sudo subscription-manager repos \
+       --enable rhocp-4.12-for-rhel-8-$(uname -i)-rpms \
+       --enable fast-datapath-for-rhel-8-$(uname -i)-rpms
+}
+
+function download_microshift_rpm {
+    local pkgDir=$1
+    pushd ${pkgDir}
+    brew download-build --quiet --arch noarch --arch ${ARCH} ${MICROSHIFT_NVR}
+    popd
+}
+
+function create_iso {
+    local pkgDir=$1
+    rm -fr microshift
+    git clone -b release-4.12 https://github.com/openshift/microshift.git
+    pushd microshift
+    sed -i '/# customizations/,$d' scripts/image-builder/config/blueprint_v0.0.1.toml
+    cat << EOF >> scripts/image-builder/config/blueprint_v0.0.1.toml
+[[packages]]
+name = "microshift-release-info"
+version = "*"
+EOF
+    sed -i 's/redhat/core/g' scripts/image-builder/config/kickstart.ks.template
+    sed -i '/--bootproto=dhcp/a\network  --hostname=api.crc.testing' scripts/image-builder/config/kickstart.ks.template
+    scripts/image-builder/cleanup.sh -full
+    # The home dir and files must have read permissions to group
+    # and others because osbuilder is running from another non-priviledged user account
+    # and allow it to read the files on current user home (like reading yum repo which is created as part of build script), it is required.
+    # https://github.com/openshift/microshift/blob/main/scripts/image-builder/configure.sh#L29-L32
+    chmod 0755 $HOME
```

Comment:
fwiw, an alternative for these chmods is something like:
```
+    setfacl -m u:_osbuild-composer:x ${HOME}
+    setfacl -m u:_osbuild-composer:x ${HOME}/snc
+    setfacl -m u:_osbuild-composer:x $(pwd)
+    mkdir _output
+    setfacl --default -m u:_osbuild-composer:rx $(pwd)/_output
+    setfacl --recursive -m u:_osbuild-composer:rx ${microshift_pkg_dir}
```

Reply:
Improvements; let's create a new PR for this

Reply:
I don't even think it's a lot better than the `chmod` to be honest, just sharing alternatives here.